### ISSUE TITLE
No Test Event for HubSpot Trigger - New Deal Property Change [BUG] 

### ISSUE THREAD

**Issue by ambo611** [PipedreamHQ/pipedream#6348]
**Describe the bug**
Created a trigger for when a Deal Property in HubSpot changes but there are no test events pulling through

**Steps to reproduce the behavior:**
1. Add new trigger
2. Select HubSpot app
3. Select New Deal Property Change
4. Choose HubSpot account
5. Set time interval
6. Choose Property ID
7. Create Source
8. Update Property value referenced in step 6 in HubSpot
9. See error - no event found 
10. Try clicking "Try Now"
11. See error - no event found 

<img width="735" alt="image" src="https://user-images.githubusercontent.com/132928443/236871341-358aa682-dec2-4252-9bc3-cbb79ce9dcdb.png">

**Desktop (please complete the following information):**
 - OS: iOS
 - Browser: Chrome
 - Version: not sure

**Comment by ambo611** [comment]
Just to add some more context

- I tried with different properties and none of them work
- Other HubSpot trigger (such as New Deal In Stage) is currently working working

**Comment by vunguyenhung** [comment]
Hi @ambo611, I can reproduce your issue as following:
- I added a new property to my deal `Client Interview Today` with Checkbox type, then I created a Pipedream source as follow
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/42bb2c91-d748-45f0-bf00-22d5fc9858d3)
- Then I set value for some deals
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/62fd80a5-9374-4f19-a115-cbfe6bc48ec9)
- Running the source manually, I did not see new events 
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/042d4f62-9767-47b6-942a-73148e2967da)
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/dda4701e-a342-46cd-8a88-40009826e7b7)

I will add this to Pipedream prioritized backlog

**Comment by ambo611** [comment]
Thank you @vunguyenhung 

I just got an email saying all of my credits for the day have been used, however, there should not have been anything triggered. I looked at the source of what was said to use the credits and it was this source.

![image](https://github.com/PipedreamHQ/pipedream/assets/132928443/eff18fce-fc20-4768-aaa6-50e577568c1c)
![image](https://github.com/PipedreamHQ/pipedream/assets/132928443/9d8197b1-3429-40f6-8178-0ee66c92d3c4)

**Comment by vunguyenhung** [comment]
> I just got an email saying all of my credits for the day have been used, however, there should not have been anything triggered. I looked at the source of what was said to use the credits and it was this source.

Hi @ambo611, a credit is used every time your trigger execute (.e.g every 15 minutes) regardless the number of event emitted during that time. You can read through this document to understand Pipedream credit https://pipedream.com/docs/pricing/#credits

**Comment by vunguyenhung** [comment]
Yes @ambo611, for now it needs to be on a timer. You can extend the interval to 30 minutes - 1 hour in order to reduce your usage of credit

**Comment by dannyroosevelt** [comment]
FYI @ambo611 if your source is what's considered a ["dependent" source](https://pipedream.com/docs/workflows/steps/triggers/#dependent-and-independent-sources) (ie, you create it from within a workflow and it's connected to only that workflow), then you should not be charged for any of those source invocations in our [new billing model](https://pipedream.com/docs/pricing/#credits).

**Comment by dannyroosevelt** [comment]
@ambo611 can you let me know the name of your workspace and can you send me a link to the source? I'd like to confirm the above is true in this case and there isn't some other issue.

**Comment by ambo611** [comment]
@dannyroosevelt my workspace is redbeardtalent and [here](https://pipedream.com/sources/dc_RWuA782) is the source: New Client Interview Today

**Comment by ambo611** [comment]
@dannyroosevelt I believe it should be a dependent source as it is not used in any other workflow

Also, there seems to be a test event now, however, there is no actual deal information retrieved 
![image](https://github.com/PipedreamHQ/pipedream/assets/132928443/0e7591f5-16f0-448b-9eef-7b26a85f86c2)

**Comment by michelle0927** [comment]
The New Deal Property Change source appears to be working for me. @ambo611 Would you mind trying it one more time after your quota issue is resolved?

<img width="815" alt="Screenshot 2023-05-11 at 3 57 28 PM" src="https://github.com/PipedreamHQ/pipedream/assets/10889229/1d9a2346-52f2-4b42-9501-56e9707a5482">

**Comment by ambo611** [comment]
@michelle0927 hmm it still seems to not be pulling any events for me

**Comment by michelle0927** [comment]
I noticed that New Deal Property Change wasn't paginating through all the deals, so I think it could be missing events for users with more than 50 deals. I created a new PR (https://github.com/PipedreamHQ/pipedream/pull/6441) to add pagination to it. @vunguyenhung It would be great if you could give it a quick check when you have a chance, to see if I'm on the right track.

**Comment by vunguyenhung** [comment]
I can confirm that the source is working for me now
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/cbd38924-b0c6-4532-8e76-1183a76b53b4)

I have over 1000 deals in Hubspot, so I think the pagination has been fixed
![image](https://github.com/PipedreamHQ/pipedream/assets/18277920/b1e7fa8e-3c2c-4357-ac0e-67b6069e8331)

**Comment by ambo611** [comment]
> @dannyroosevelt thank you! my workspace is redbeardtalent and [here](https://pipedream.com/sources/dc_RWuA782) is the source: New Client Interview Today

@dannyroosevelt Is this a dependent or independent source? 

For reference,
workspace: redbeardtalent
source: [New Client Interview Today Property Change](https://pipedream.com/sources/dc_BVuzaXg)

also curious about
Source: [New Application Shortlisted](https://pipedream.com/sources/dc_PnuZgOD)

**Comment by dannyroosevelt** [comment]
I see `dc_BVuzaXg` as `dependent` on our end -- can you confirm what you're seeing?

I cannot find any info for `dc_PnuZgOD` -- can you doublecheck that ID?
